Warning free compilation

php1ic opened the pull request:
Changes to get the project compiling without any warnings. Only one left.

I don't know enough Qt to quickly fix this one. Will keep working on it.
```
/home/ijc/repo/nuclei/source/Nuclei.cpp: In member function ‘void Nuclei::initialize()’:
/home/ijc/repo/nuclei/source/Nuclei.cpp:105:20: warning: ‘QModelIndex QModelIndex::child(int, int) const’ is deprecated: Use QAbstractItemModel::index [-Wdeprecated-declarations]
  105 |       mi = mi.child(selectionIndices.at(i).toInt(), 0);
      |            ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt5/QtWidgets/qabstractitemview.h:45,
                 from /usr/include/qt5/QtWidgets/qheaderview.h:44,
                 from /usr/include/qt5/QtWidgets/QHeaderView:1,
                 from /home/ijc/repo/nuclei/build/source/nuclei_autogen/include/ui_Nuclei.h:18,
                 from /home/ijc/repo/nuclei/source/Nuclei.cpp:2:
/usr/include/qt5/QtCore/qabstractitemmodel.h:455:20: note: declared here
  455 | inline QModelIndex QModelIndex::child(int arow, int acolumn) const
      |                    ^~~~~~~~~~~

```

php1ic committed:
Fix deprecated-copy warnings

The majority of the warnings came from definition of = operator which
is not required if the user has defined a copy constructor.

php1ic committed:
Fix unused parameter warnings

It looks like most of these were from debug, or previously used
functionality. Until I know more about the project, most have been
commented out rather than deleted.

php1ic committed:
Fix deprecated-declaration warnings

I don't know Qt so have just followed the compiler warning advice and
confirmed the code still runs.

ferdymercury commented:
> don't know enough Qt to quickly fix this one. Will keep working on it.

Maybe this helps: https://phabricator.kde.org/D23959

Thanks for your work!!

php1ic committed:
More deprecated-declaration warning fixes

I forgot to clean the build directory before the previous commit so
missed a few.

php1ic commented:
Forgot to clean before building last time so missed a few. Definitely only 1 warning left now.
```
/home/ijc/repo/nuclei/source/ENSDFDataSource.cpp: In member function ‘QList<short unsigned int> ENSDFDataSource::getAvailableDataFileNumbers()’:
/home/ijc/repo/nuclei/source/ENSDFDataSource.cpp:149:38: warning: ‘static QList<T> QList<T>::fromStdList(const std::__cxx11::list<T>&) [with T = short unsigned int]’ is deprecated: Use QList<T>(list.begin(), list.end()) instead. [-Wdeprecated-declarations]
  149 |   return QList<uint16_t>::fromStdList(parser.masses());
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
In file included from /usr/include/qt5/QtCore/qstringlist.h:41,
                 from /usr/include/qt5/QtCore/QStringList:1,
                 from /home/ijc/repo/nuclei/source/ENSDFDataSource.h:3,
                 from /home/ijc/repo/nuclei/source/ENSDFDataSource.cpp:1:
/usr/include/qt5/QtCore/qlist.h:416:28: note: declared here
  416 |     static inline QList<T> fromStdList(const std::list<T> &list)
      |                            ^~~~~~~~~~~
```
However following the advice and using the new constructor slows startup time to a crawl
```
return QList<uint16_t> {parser.masses().begin(), parser.masses().end()};
```
I gave up after ~5mins of waiting for the UI to appear.

https://doc.qt.io/qt-5/qlist-obsolete.html#fromStdList

php1ic committed:
Upgrade versions of spdlog and fmt

php1ic committed:
Use all available cores to build in the CI pipeline

martukas commented on source/SchemeEditor/TransitionItem.cpp:
```diff
@@ -126,7 +126,7 @@ Energy TransitionItem::to() const
 }
 
-void TransitionItem::updateArrow(double arrowDestY, double max_intensity)
+void TransitionItem::updateArrow(double arrowDestY /*, double max_intensity*/)
```
Instead of commenting out, maybe use this?:
https://en.cppreference.com/w/cpp/language/attributes/maybe_unused

martukas commented on source/SchemeEditor/TransitionItem.h:
```diff
@@ -27,7 +27,7 @@ class TransitionItem : public ClickableItem
   virtual ~TransitionItem();
 
-  void updateArrow(double arrowDestY, double max_intensity);
+  void updateArrow(double arrowDestY /*, double max_intensity*/);
```
Ditto here, may be other spots I did not notice.

martukas commented on source/ensdf/records/Continuation.cpp:
```diff
@@ -222,27 +222,27 @@ parse_continuation(const std::string& crecs)
-                         const std::map<std::string, Continuation>& from,
-                         const std::string &debug_line)
+                         const std::map<std::string, Continuation>& from)
+                         //const std::string &debug_line)
```
And here

martukas commented on source/ensdf/records/Continuation.cpp:
```diff
@@ -222,27 +222,27 @@ parse_continuation(const std::string& crecs)
-      if (to[cont.first].is_uncert()
-          && cont.second.is_uncert())
-      {
+    // else
```
I would err in the other direction. Commented code tends to rot.
Maybe switch the print statement below from DBG to TRC to spam less?

php1ic replied:
The call has been changed to TRC, but the logger doesn't know how to format the Continuation struct so I have left it commented while I look into that.

martukas replied:
Yeah, you probably need to either write an `fmt` interface for it or just some sort of `to_string()` function.

php1ic replied:
Custom formatter added with a best guess for what should be printed.

martukas reviewed (commented):
Thank you for the contribution! A few suggestions.

php1ic committed:
Change getter function to return a QList

The old getter was converting a set to a list then that list was
immediately being converted to a QList. This additional step looks to
be the cause of the slow start up when using the new QList constructor
and moving away from the deprecated fromstdlist.

php1ic committed:
Put the unused parameters back in but mark as maybe_unused

It looks as if they were used in the past so might be used again so
for the moment, keep them in but use the maybe_unused attribute to
remove compilation warnings.

The logging function don't know how to format the Continuation struct
so leave the updated call commented for the moment.

php1ic commented:
Start up time has been fixed and unused parameters uncommented and marked as maybe_unused.

There is an issue with the TRC logging function not knowing how to format the Continuation struct so I have updated the code but left the call commented to be fixed another time.

php1ic committed:
Add custom formatter for the Continuation struct

php1ic committed:
Add an extension to the debug log file and tell git to ignore it

php1ic committed:
Ignore the directories automatically created by Qt

martukas reviewed (approved):
LGTM

php1ic commented:
My linter is still showing the odd warning, but compilation now has no warnings so I think this PR can be closed.

I'm not able to merge this, can one of you please merge it into the main branch?
